parser_library/src/processing/statement_analyzers/hit_count_analyzer.cpp

```diff
@@ -28,22 +29,27 @@ hit_count_analyzer::hit_count_analyzer(context::hlasm_context& ctx)
 {}
 
 namespace {
-std::optional<range> get_range(const context::hlasm_statement* stmt,
+static constexpr auto stmt_lines_compacter = [](const range& r) { return std::make_pair(r.start.line, r.end.line); };
+
+std::optional<stmt_lines_range> get_stmt_lines_range(const context::hlasm_statement* stmt,
     statement_provider_kind prov_kind,
     processing_kind proc_kind,
     const context::hlasm_context& ctx)
 {
     if (!stmt)
         return std::nullopt;
 
-    const auto adj_range = [prov_kind, &proc_kind, &ctx](range r) -> std::optional<range> {
+    const auto adj_range = [prov_kind, &proc_kind, &ctx](const range& r) -> std::optional<stmt_lines_range> {
         if (proc_kind != processing_kind::LOOKAHEAD && r.start == r.end)
             return std::nullopt;
 
         if (prov_kind == statement_provider_kind::OPEN)
-            r.end.line = ctx.current_source().end_index - 1;
-        r.end.column = 72;
-        return r;
+        {
+            const auto& source = ctx.current_source();
+            return std::make_pair(r.start.line, r.end.line + source.end_index - source.begin_index - 1);
```

Comment:
I did not realize that opencode statements can have ranges that are this off... Looks like something worth looking into later on.

Reply:
`TODO` added